tests/e2e/e2e_test.go

```diff
@@ -0,0 +1,276 @@
+		ccancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+		// all logs are stored under root data dir
+		logsDir = resp.GetClusterInfo().GetRootDataDir()
+
+		if v, ok := resp.ClusterInfo.CustomVms[k.String()]; ok {
+			blockchainID = v.BlockchainId
+			outf("{{blue}}subnet-evm is ready:{{/}} %+v\n", v)
+			break done
+		}
+	}
+	gomega.Expect(ctx.Err()).Should(gomega.BeNil())
+	cancel()
+
+	gomega.Expect(blockchainID).Should(gomega.Not(gomega.BeEmpty()))
+	gomega.Expect(logsDir).Should(gomega.Not(gomega.BeEmpty()))
+
+	cctx, ccancel := createDefaultCtx()
+	uris, err := cli.URIs(cctx)
+	ccancel()
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("{{blue}}avalanche HTTP RPCs URIs:{{/}} %q\n", uris)
+
+	for _, u := range uris {
+		rpcEP := fmt.Sprintf("%s/ext/bc/%s/rpc", u, blockchainID)
+		subnetEVMRPCEps = append(subnetEVMRPCEps, rpcEP)
+		outf("{{blue}}avalanche subnet-evm RPC:{{/}} %q\n", rpcEP)
+	}
+
+	pid := os.Getpid()
+	outf("{{blue}}{{bold}}writing output %q with PID %d{{/}}\n", outputPath, pid)
+	ci := clusterInfo{
+		URIs:     uris,
+		Endpoint: fmt.Sprintf("/ext/bc/%s", blockchainID),
+		PID:      pid,
+		LogsDir:  logsDir,
+	}
+	gomega.Expect(ci.Save(outputPath)).Should(gomega.BeNil())
+
+	b, err := os.ReadFile(outputPath)
+	gomega.Expect(err).Should(gomega.BeNil())
+	outf("\n{{blue}}$ cat %s:{{/}}\n%s\n", outputPath, string(b))
+})
+
+var _ = ginkgo.AfterSuite(func() {
+	switch mode {
+	case "test":
+		outf("{{red}}shutting down cluster{{/}}\n")
+		ctx, cancel := createDefaultCtx()
+		_, err := cli.Stop(ctx)
+		cancel()
+		gomega.Expect(err).Should(gomega.BeNil())
+
+	case "run":
+		outf("{{red}}skipping shutting down cluster{{/}}\n")
+	}
+
+	outf("{{red}}shutting down client{{/}}\n")
+	gomega.Expect(cli.Close()).Should(gomega.BeNil())
+})
+
+var _ = ginkgo.Describe("[basic]", func() {
+	ginkgo.It("can TODO", func() {
+		if mode != "test" {
+			ginkgo.Skip("mode is not 'test'; skipping...")
+		}
+
+		// TODO: e2e tests specific for subnet-evm
+	})
+})
+
+func createDefaultCtx() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), 2*time.Minute)
+}
+
+// Outputs to stdout.
+//
+// e.g.,
+//   Out("{{green}}{{bold}}hi there %q{{/}}", "aa")
+//   Out("{{magenta}}{{bold}}hi therea{{/}} {{cyan}}{{underline}}b{{/}}")
+//
+// ref.
+// https://github.com/onsi/ginkgo/blob/v2.0.0/formatter/formatter.go#L52-L73
+//
+func outf(format string, args ...interface{}) {
+	s := formatter.F(format, args...)
+	fmt.Fprint(formatter.ColorableStdOut, s)
+}
+
+// clusterInfo represents the local cluster information.
+type clusterInfo struct {
+	URIs     []string `json:"uris"`
+	Endpoint string   `json:"endpoint"`
+	PID      int      `json:"pid"`
+	LogsDir  string   `json:"logsDir"`
+}
+
+const fsModeWrite = 0o600
+
+func (ci clusterInfo) Save(p string) error {
+	ob, err := yaml.Marshal(ci)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(p, ob, fsModeWrite)
+}
+
+func getVMID(vmName string) (ids.ID, error) {
```

Comment:
Can we add this as a utility within AvalancheGo? We should probably use this as a standard practice to create our VMIDs instead of having a private helper function within this package.

Reply:
Left as TODO, will add a util to avalanchego and import here in the following PR.